lib/explorer/data_frame.ex

```diff
@@ -5315,6 +5315,69 @@ defmodule Explorer.DataFrame do
     groups ++ names_with_dtypes
   end
 
+  @doc """
+  Explodes one or multiple list column into multiple rows.
+
+  When exploding multiple columns, the number of list elements in each row for
+  the exploded columns must be the same.
+
+  ## Examples
+
+      iex> df = Explorer.DataFrame.new(a: [[1, 2], [3, 4]], b: [[5, 6], [7, 8]], c: ["a", "b"])
+      iex> Explorer.DataFrame.explode(df, :a)
+      #Explorer.DataFrame<
+        Polars[4 x 3]
+        a integer [1, 2, 3, 4]
+        b list[integer] [[5, 6], [5, 6], [7, 8], [7, ...]]
+        c string ["a", "a", "b", "b"]
+      >
+      iex> Explorer.DataFrame.explode(df, [:a, :b])
+      #Explorer.DataFrame<
+        Polars[4 x 3]
+        a integer [1, 2, 3, 4]
+        b integer [5, 6, 7, 8]
+        c string ["a", "a", "b", "b"]
+      >
+
+  You can think of exploding multiple list columns as being the inverse of
+  aggregating the elements the exploded columns into lists:
+
+      iex> df = Explorer.DataFrame.new(a: [1, 2, 3, 4], b: [5, 6, 7, 8], c: ["a", "a", "b", "b"])
+      iex> df = df |> Explorer.DataFrame.group_by(:c) |> Explorer.DataFrame.summarise(a: a, b: b)
+      #Explorer.DataFrame<
+        Polars[2 x 3]
+        c string ["a", "b"]
+        a list[integer] [[1, 2], [3, 4]]
+        b list[integer] [[5, 6], [7, 8]]
+      >
+      iex> Explorer.DataFrame.explode(df, [:a, :b]) # we are back where we started
+      #Explorer.DataFrame<
+        Polars[4 x 3]
+        c string ["a", "a", "b", "b"]
+        a integer [1, 2, 3, 4]
+        b integer [5, 6, 7, 8]
+      >
+
+  If you want to perform the cartesian product of two list columns, you must
+  call `explode/2` once for each column:
+
+      iex> df = Explorer.DataFrame.new(a: [[1, 2], [3, 4]], b: [[5, 6], [7, 8]], c: ["a", "b"])
+      iex> df |> Explorer.DataFrame.explode(:a) |> Explorer.DataFrame.explode(:b)
+      #Explorer.DataFrame<
+        Polars[8 x 3]
+        a integer [1, 1, 2, 2, 3, ...]
+        b integer [5, 6, 5, 6, 7, ...]
+        c string ["a", "a", "a", "a", "b", ...]
+      >
+  """
+  @doc type: :single
+  @spec explode(df :: DataFrame.t(), column :: column_name() | [column_name()]) :: DataFrame.t()
+  def explode(%DataFrame{} = df, column_or_columns) do
+    columns = to_existing_columns(df, List.wrap(column_or_columns))
```

Comment:
It just silently ignores the non-list columns and the operation is successful. Do you think we should raise on the Elixir side in this case?

Reply:
@costaraphael personally I think it should raise if one of the columns is not of a list dtype.

Reply:
I agree!

Reply:
@philss @josevalim done in [a1ad581](https://github.com/elixir-explorer/explorer/pull/751/commits/a1ad5812eb0cb24fde3ca1dd62904fd18acb8d04)!

Since it seems like this is the first time this kind of validation is performed at the `DataFrame` level (checking if the given columns are of a certain type), I've chosen not to introduce any new abstractions. I figured this could be done once there were a couple more examples of this. Let me know if you disagree!